edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/EdgeDeploymentControllerTest.cs

```diff
@@ -0,0 +1,590 @@
+            string imageName = "test-image:1";
+            KubernetesModule km1 = this.CreateKubernetesModule(moduleName, creatOptions, imageName);
+            string newImage = "test-image:2";
+            KubernetesModule km1UpdaModule = this.CreateKubernetesModule(moduleName, creatOptions, newImage);
+            var labels = new Dictionary<string, string>
+            {
+                ["net.azure-devices.edge.deviceid"] = "deviceid",
+                ["net.azure-devices.edge.hub"] = "hostname",
+                ["net.azure-devices.edge.module"] = moduleName
+            };
+
+            await this.client.AddModuleDeploymentAsync(moduleName, labels, null);
+            moduleLifeCycleManager.SetModules(moduleName);
+            await controller.DeployModulesAsync(ModuleSet.Create(km1UpdaModule), ModuleSet.Create(km1));
+            await this.client.ReplaceModuleImageAsync(moduleName, newImage);
+            await controller.DeployModulesAsync(ModuleSet.Create(km1UpdaModule), ModuleSet.Create(km1));
+
+            V1DeploymentList currentDeployments = await this.client.ListDeployments(deviceSelector);
+            V1ServiceAccountList currentServiceAccounts = await this.client.ListServiceAccounts(deviceSelector);
+            V1ServiceList currentServices = await this.client.ListServices(deviceSelector);
+            V1PersistentVolumeClaimList currentPvcList = await this.client.ListPeristentVolumeClaims();
+            Assert.Single(currentDeployments.Items, d => d.Spec.Template.Spec.Containers[0].Image == newImage);
+            Assert.Single(currentServiceAccounts.Items, sa => sa.Metadata.Name == moduleName);
+            Assert.Empty(currentServices.Items);
+            Assert.Empty(currentPvcList.Items);
+        }
+
+        public class DummyModuleIdentityLifecycleManager : IModuleIdentityLifecycleManager
+        {
+            readonly string hostName;
+            readonly string gatewayHostname;
+            readonly string deviceId;
+            readonly string moduleId;
+            readonly ICredentials credentials;
+            private IImmutableDictionary<string, IModuleIdentity> identites = ImmutableDictionary<string, IModuleIdentity>.Empty;
+            public DummyModuleIdentityLifecycleManager(string hostName, string gatewayHostname, string deviceId, string moduleId, ICredentials credentials)
+            {
+                this.hostName = hostName;
+                this.gatewayHostname = gatewayHostname;
+                this.deviceId = deviceId;
+                this.moduleId = moduleId;
+                this.credentials = credentials;
+            }
+
+            public Task<IImmutableDictionary<string, IModuleIdentity>> GetModuleIdentitiesAsync(ModuleSet desired, ModuleSet current) => Task.FromResult(this.identites);
+
+            IModuleIdentity CreateModuleIdentity() => new ModuleIdentity(this.hostName, this.gatewayHostname, this.deviceId, this.moduleId, this.credentials);
```

Comment:
this method should be below the method SetModules. because SetModules uses CreateModuleIdentity, and not vise versa